### PR TITLE
upgrade ctf

### PR DESCRIPTION
**Pull request by jmank88**
image

### CODE REVIEW

File: integration-tests/go.mod [hunk 1]
```diff
@@ -455,37 +473,44 @@ require (
...
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-	// until merged upstream: https://github.com/mwitkow/grpc-proxy/pull/69
-	github.com/mwitkow/grpc-proxy => github.com/smartcontractkit/grpc-proxy v0.0.0-20230731113816-f1be6620749f
+	// K8s imports are weird
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/common => github.com/prometheus/common v0.42.0
+
+	// type func(a Label, b Label) bool of func(a, b Label) bool {…} does not match inferred type func(a Label, b Label) int for func(a E, b E) int
+	github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.47.2-0.20231010075449-4b9c19fe5510
```
Comment (jmank88) [inline integration-tests/go.mod:513]: Copied from chainlink-solana to make this compile :disappointed:

**Review by aalu1418 — COMMENTED** [review]
there's some changes that also need to be ported over too

here's a few of them (not sure if this is comprehensive)
- https://github.com/smartcontractkit/chainlink-solana/pull/842 (this is the ctf => ctf/lib change which is already done)
- https://github.com/smartcontractkit/chainlink-solana/pull/840 (some method changes)

cc @augustbleeds

**Comment by augustbleeds** [comment]
Hey 👋 , what's the context of these changes and what's ctf?

For the PRs that @aalu1418 is mentioning, are they needed before this PR merged in or just required at some later date.

**Comment by jmank88** [comment]
Just `TestOCRBasic` failing:
```
13:58:56.72 DBG Validating test config
      error validating test config: chainlink image config validation failed: chainlink image name must be set
```

**Comment by aalu1418** [comment]
CTF is chainlink-testing-framework
the changes i mentioned needed can be added within this PR otherwise it will continue to fail

did this the testconfig changes ever get ported to starknet? https://github.com/smartcontractkit/chainlink-solana/pull/817
this might also be needed
cc @lukaszcl

**Comment by jmank88** [comment]
> CTF is chainlink-testing-framework the changes i mentioned needed can be added within this PR otherwise it will continue to fail

Am I missing something?

**Comment by aalu1418** [comment]
> Am I missing something?

potentially the testconfig changes? but it could just be the .github workflow changes that are in the PR i linked 🤔

**Comment by jmank88** [comment]
> potentially the testconfig changes? but it could just be the .github workflow changes that are in the PR i linked 🤔

I don't see an equivalent workflow in this repo, but I was missing the field change in the nil check :crossed_fingers:

**Comment by jmank88** [comment]
Still failing the same way. Where do we set the chainlink image?

**Comment by aalu1418** [comment]
try making this change in https://github.com/smartcontractkit/chainlink-starknet/blob/develop/.github/workflows/integration-tests-smoke.yml 
image

**Review by skudasov — DISMISSED** [review]

**Review by skudasov — APPROVED** [review]

**Pull request merged**